firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt

```diff
@@ -79,7 +86,27 @@ internal constructor(
       apiKey,
       modelName,
       requestOptions.toInternal(),
-      "gl-kotlin/ fire/${BuildConfig.VERSION_NAME}"
+      "gl-kotlin/ fire/${BuildConfig.VERSION_NAME}",
+      object : HeaderProvider {
+        override val timeout: Duration
+          get() = 10.seconds
+
+        override suspend fun generateHeaders(): Map<String, String> {
+          if (appCheckTokenProvider == null) {
+            Log.w(TAG, "AppCheck not registered, skipping")
+            return emptyMap()
+          }
+          val mutableHeaderMap: MutableMap<String, String> = mutableMapOf()
+
+          val token = appCheckTokenProvider.getToken(false).await()
+          if (token.error != null) {
+            Log.w(TAG, "Error obtaining appcheck token", token.error)
+          }
+          mutableHeaderMap.put("X-Firebase-AppCheck", token.token)
```

Comment:
Will `token` be present if `token.error` is? Shouldn't you have a return in the above block?

Reply:
Yeah, that's one peculiarity of appcheck, token will be present even if there's an error. 

https://github.com/firebase/firebase-android-sdk/blob/a7edffefa9cd4f45b5921db3fd2003101c38e2ef/appcheck/firebase-appcheck-interop/src/main/java/com/google/firebase/appcheck/AppCheckTokenResult.java#L21